components/shared/MarkdownEditor.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import styled, { css } from 'styled-components';
+import parse from 'html-react-parser';
+
+import SquareButton from './SquareButton';
+import Icon from './Icon';
+
+const MarkdownIt = require('markdown-it');
+
+const Toggler = styled(SquareButton)`
+  float: right;
+  margin-bottom: 0.25rem;
+  width: 7rem;
+  height: 3rem;
+`;
+
+const shared = css`
+  border: 1px solid ${({ theme }) => theme.colors.mediumGray};
+  width: 100%;
+  max-width: inherit;
+  min-height: 20rem;
+  overflow: auto;
+  padding: 0.5rem;
+  margin: 0;
+`;
+
+const TextArea = styled.textarea`
+  ${shared}
+  resize: vertical;
+`;
+
+const Preview = styled.div`
+  ${shared}
+  min-height: 20rem;
+`;
+
+const MarkdownNote = styled.p`
+  margin: 0;
+  float: right;
+  margin-top: -1rem;
+  font-size: 1.2rem;
+`;
+
+const converter = new MarkdownIt();
+let formik;
+let fieldName;
+let invalid = false;
+let markdown = '';
+let preview = '';
+let state = 'edit';
+
+const handleStateToggle = () => {
+  if (state === 'edit') {
```

Comment:
Would it help to use `useState` here? You could even go with a `displayPreview` state to use a boolean, make it easy to toggle with `setState`. An idear.

Reply:
not sure it would help anything and was trying to avoid using state for a single field, but if you think it would be cleaner I can change it.  Let me know your thoughts.

Reply:
I tend to lean towards state as it seems to clean up the logic some. Even for the tiny state changes like this. Going with a boolean for it too you can easily toggle the state without the if/else here, so you'd get some clean up here and then can pull into into the checks in your render. Thinking a la...
```
{state === 'preview' && <Preview>{parse(preview)}</Preview>}
```
to:
```
{displayPreview && <Preview>{parse(converter.render(markdown))}</Preview>}
```

Reply:
@saragibby updated code to use state